src/System.Management.Automation/System.Management.Automation.csproj

```diff
@@ -38,6 +56,11 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <!-- exclude code of source generators from compilation and IDEs (e.g. vscode and visual studio) -->
+    <Compile Remove="SourceGenerators\**\*.cs" />
+    <!-- exclude the generated source files because compiler already includes the in-memory source generator output -->
+    <Compile Remove="$(CompilerGeneratedFilesOutputPath)\**\*.cs" />
```

Comment:
The only remaining problem is: the generated source code is not visible to IDEs like VSCode and Visual Studio.
The `<Compile Remove ... />` tag is needed because otherwise the compiler will report errors about duplicate declarations, as it automatically includes the im-memory source output, as well as the file generated from the last build.

In Visual Studio or VSCode, the constant members used in `PSVersionInfo.cs` cannot be resolved, because when using `<Compile Remove ... />`, the files will not be included for analyzing. I tried many ways, including attempting to delete the generated file using `<RemoveDir>` task in `System.Management.Automation.csproj` or `PSVersionInfoGenerator.csproj`, but nothing works. @iSazonov and @SeeminglyScience, really need help on this.

Reply:
Eh, it's been so long that I don't remember all the details anymore. Yes, there was such a problem and this line was a workaround. I even opened the case https://github.com/OmniSharp/omnisharp-vscode/issues/4215 and there's a progression in recent months, but I haven't checked if it's already in the current public version of VSCode/OmniSharp.
I think I saw in the VS 2022 announcement that it has SGs support, but I haven't checked and can't say for sure.

I'll try to find something useful

Reply:
I added cleanup commit:
- Turn off EmitCompilerGeneratedFiles - since latest VS Code works well we have no need to keep source of the generated file.
- Remove first blank line in generated code - cosmetic change

Reply:
It doesn't work in the following scenarios, for both VSCode and Visual Studio, which are quite common in development:
1. Run `Start-PSBuild` first, then open `System.Management.Automation` in VSCode
2. Open `System.Management.Automation` in VSCode to use an existing build, like from yesterday.

But I'm fine turning off `EmitCompilerGeneratedFiles` since it doesn't help. Also, I'm fine merging the PR without perfectly resolving this issue.

Reply:
I pushed a new commit to solve this issue, not perfectly, but works.

Reply:
I think this is fundamentally unsolvable because the code is generated on the fly and exists only at the moment of compilation. In other words OmniSharp must already be running to capture this code. 

> Also, I'm fine merging the PR without fully resolving this issue.

It's up to you. I don't mind if this workaround remains.